### PR TITLE
Fix #5 dump http headers

### LINKED ISSUE #5: Add LOG_HTTP_HEADERS option to enable logging of the request headers.
This would be similar to the existing LOG_HTTP_BODY environment variable.

### PR DESCRIPTION
**Pull request by arulrajnet**
- multistage build. So no need to install any dependencies

- docker-compose.yml for bringup and local test

- readme updated

image

### CODE REVIEW

**Comment by jmalloc** [comment]
Hi, thanks for the PR!

The changes around `LOG_HTTP_HEADERS` look good, and I would be happy to merge them. However, the way the `Dockerfile` has been changed doesn't work quite right with my Makefile setup. I realise the README is lacking in this regard, but `make docker` is all that's needed to build the Docker image locally.

**Comment by arulrajnet** [comment]
@jmalloc The current `Dockerfile` assumes the golang has to be there to build the binary. I have changed that as multistage build https://docs.docker.com/develop/develop-images/multistage-build/ . Compiling the binary inside the docker container and copy into actual image. 

To compile this we need Golang 1.13+ . I have older golang version for my project requirement. Don't want to upgrade golang for this. That's why taken this approach. Thought of everyone will face similar issue while contributing. So committed.  

I will remove this If you strongly believe this is not needed.

BTW to build docker

```
docker build -t jmalloc/echo-server .
```

Already updated in readme

**Comment by arulrajnet** [comment]
@jmalloc For the purpose of testing I have upgraded my go. I have tested the `make docker` with go version `1.14.2`. Everything works fine to me. 

image

What is the issue you are facing? Could you give me the error log?

One of my automate testing depends on this. Please publish image to docker once merged. Thanks.

**Comment by jmalloc** [comment]
There is no error, as such, it's just that this change uses the Makefiles in away that they were not designed.

There are several ways that this could be problematic, but just to give one simple example, `make docker` now results in two separate compilations of the binary, once locally and again under Docker.  The expectation when using these Makefiles is that the binary built by make as a prerequisite of the `docker` recipe is the one that will be copied into the Docker image, which is no longer true.

In short, the Makefiles are "in charge" and I avoid involving Docker in the build process any more than is necessary to build the Docker image, which is just one possible artefact of the build. 

FWIW, While I can see this might not pose an immediate problem for this specific project because it's not being actively developed (and has no tests 😱 ), this Makefile system is used widely across several organisations by me and my colleagues and I strongly prefer to avoid special cases.

**Comment by arulrajnet** [comment]
@jmalloc updated the same.

**Comment by jmalloc** [comment]
Thanks! v0.2.0 has been pushed to Docker hub.

https://hub.docker.com/layers/jmalloc/echo-server/0.2.0/images/sha256-0a93426f1960939c3952e3f7a1a3cda0f8d529b0bc0ee036edc0222d6e33ee5e?context=explore
